src/oneD/StFlow.cpp

```diff
@@ -344,190 +354,259 @@ void StFlow::updateProperties(size_t jg, double* x, size_t jmin, size_t jmax)
-        double boundary_Rad_right = m_epsilon_right * StefanBoltz * pow(T(x, m_points - 1), 4);
-
-        // loop over all grid points
-        for (size_t j = jmin; j < jmax; j++) {
-            // helping variable for the calculation
-            double radiative_heat_loss = 0;
-
-            // calculation of the mean Planck absorption coefficient
-            double k_P = 0;
-            // absorption coefficient for H2O
-            if (m_kRadiating[1] != npos) {
-                double k_P_H2O = 0;
-                for (size_t n = 0; n <= 5; n++) {
-                    k_P_H2O += c_H2O[n] * pow(1000 / T(x, j), (double) n);
-                }
-                k_P_H2O /= k_P_ref;
-                k_P += m_press * X(x, m_kRadiating[1], j) * k_P_H2O;
+    // Variable definitions for the Planck absorption coefficient and the
+    // radiation calculation:
+    double k_P_ref = 1.0*OneAtm;
+
+    // Polynomial coefficients:
+    const double c_H2O[6] = {-0.23093, -1.12390, 9.41530, -2.99880,
+                                    0.51382, -1.86840e-5};
+    const double c_CO2[6] = {18.741, -121.310, 273.500, -194.050,
+                                    56.310, -5.8169};
+
+    // Calculation of the two boundary values
+    double boundary_Rad_left = m_epsilon_left * StefanBoltz * pow(T(x, 0), 4);
+    double boundary_Rad_right = m_epsilon_right * StefanBoltz * pow(T(x, m_points - 1), 4);
+
+    for (size_t j = jmin; j < jmax; j++) {
+        // calculation of the mean Planck absorption coefficient
+        double k_P = 0;
+        // Absorption coefficient for H2O
+        if (m_kRadiating[1] != npos) {
+            double k_P_H2O = 0;
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_H2O += c_H2O[n] * pow(1000 / T(x, j), (double) n);
             }
-            // absorption coefficient for CO2
-            if (m_kRadiating[0] != npos) {
-                double k_P_CO2 = 0;
-                for (size_t n = 0; n <= 5; n++) {
-                    k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
-                }
-                k_P_CO2 /= k_P_ref;
-                k_P += m_press * X(x, m_kRadiating[0], j) * k_P_CO2;
+            k_P_H2O /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[1], j) * k_P_H2O;
+        }
+        // Absorption coefficient for CO2
+        if (m_kRadiating[0] != npos) {
+            double k_P_CO2 = 0;
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
             }
+            k_P_CO2 /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[0], j) * k_P_CO2;
+        }
+
+        // Calculation of the radiative heat loss term
+        double radiative_heat_loss = 0;
+        radiative_heat_loss = 2 * k_P *(2 * StefanBoltz * pow(T(x, j), 4)
+                              - boundary_Rad_left - boundary_Rad_right);
+
+        // set the radiative heat loss vector
+        m_qdotRadiation[j] = radiative_heat_loss;
```

Comment:
Further simplification:
```suggestion
        // Calculation of the radiative heat loss term
        m_qdotRadiation[j] = 2 * k_P * (2 * StefanBoltz * pow(T(x, j), 4)
                                        - boundary_Rad_left - boundary_Rad_right);
```